installer/src/tplink.rs

```diff
@@ -0,0 +1,297 @@
+use std::net::SocketAddr;
+use std::str::FromStr;
+use std::time::Duration;
+
+use anyhow::{Context, Error};
+use axum::{
+    Router,
+    body::{Body, to_bytes},
+    extract::{Request, State},
+    http::uri::Uri,
+    response::{IntoResponse, Response},
+    routing::any,
+};
+use bytes::{Bytes, BytesMut};
+use hyper::StatusCode;
+use hyper_util::{client::legacy::connect::HttpConnector, rt::TokioExecutor};
+use serde::Deserialize;
+use tokio::io::{AsyncReadExt, AsyncWriteExt};
+use tokio::net::TcpStream;
+use tokio::time::{sleep, timeout};
+
+use crate::InstallTpLink;
+
+type HttpProxyClient = hyper_util::client::legacy::Client<HttpConnector, Body>;
+
+pub async fn main_tplink(args: InstallTpLink) -> Result<(), Error> {
+    let InstallTpLink {
+        skip_sdcard,
+        admin_ip,
+    } = args;
+
+    let qcmap_web_cgi_endpoint = format!("http://{admin_ip}/cgi-bin/qcmap_web_cgi");
+    let client = reqwest::Client::new();
+
+    #[derive(Deserialize)]
+    struct RootResponse {
+        result: u64,
+    }
```

Comment:
i'm not a fan of declaring structs in function bodies, maybe move this to the top of the file?